lib/practice_exercises.rb

```diff
@@ -1,13 +1,34 @@
+# Kareha Agesa
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(1/2n) = O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase.gsub(/[^A-Za-z0-9]/, '')   # Space: O(n)?, Time O(n)?
+  length = string.length                              # Space: 0(1), Time O(n)
```

Comment:
Technically the Ruby `.length` method just returns the length of the string, saved as a variable.  So it has O(1).